src/main/java/io/confluent/connect/elasticsearch/ElasticsearchSinkConnectorConfig.java

```diff
@@ -369,6 +442,42 @@ private SecurityProtocol securityProtocol() {
     return SecurityProtocol.valueOf(getString(ELASTICSEARCH_SECURITY_PROTOCOL_CONFIG));
   }
 
+  public boolean isBasicProxyConfigured() {
+    return !getString(PROXY_HOST_CONFIG).isEmpty();
+  }
+
+  public boolean isProxyWithAuthenticationConfigured() {
+    return isBasicProxyConfigured()
+        && !getString(PROXY_USERNAME_CONFIG).isEmpty()
+        && getPassword(PROXY_PASSWORD_CONFIG) != null;
+  }
+
+  private void validateProxyConfigs() {
```

Comment:
Do you think we can add a validator that the proxy can be connected to?   This will validate that the user is putting in a real host, their firewall/network is correct, etc.    My concern is that if someone does something like misspell their hostname or put the wrong username/password, they will see some weird Java stacktrace, but it will be much better to see a validation error.

This would kind of be a step higher than we've done with validators in the past, but I think it will vastly improve UX and reduce confusion by users, make them dig through fewer stacktraces where it's unclear if it is user error or a true bug.

Reply:
basically the way I see this going is we try to setup a connection and ping some website i.e `confluent.io` through their proxy. This is a pretty heavy weight validation. If you have a better suggestion, I am open to it.

I think this is definitely an awesome validation for cloud specifically and we should have very heavy duty validation for as much as possible in all of the cloud connectors. Is it worth it right for on prem? Should we implement this validation as we onboard to cloud? Should we develop some sort of framework or approach to this heavy duty validation? 

I am open to implementing it now but I think then we should develop a standard for it and do it right

Reply:
why would you have to ping `confluent.io`? can't we add logic to the validate() method in the connector to make a connection to the ES server via the proxy and check if the server exists and try to do some test reads/writes?

Reply:
I can create a `new JestElasticsearchClient()` in the config which will try to connect to ES to get the version and will fail if it can't connect and that will test the entire connection configs... it seems a bit heavyweight but how does that sound?

Reply:
and the same method is used in the tasks as well? if so, then sgtm.

Reply:
turns out creating a `new JestElasticsearchClient()` will not work because it creates its own internal `Config` which results in an infinite loop of creating each other. I am opting to not do full validation that connection will be successful as that will be quite involved and am going to just file a ticket to get this done. 

The tasks already create their own client, which will fail if it can't connect.

Not sure by what you are asking, but the above method is called every time a `config` is constructed, so it will be validated immediately.